lib/validates_overlap/overlap_validator.rb

```diff
@@ -90,19 +90,27 @@ def primary_key(record)
     record.class.primary_key
   end
 
+  def primary_key_value(primary_key_name,record)
+    record.send(primary_key_name)
+  end
+
   # Generate sql condition for time range cross
   def generate_overlap_sql_conditions(record)
     starts_at_attr, ends_at_attr = attributes_to_sql(record)
     main_condition = condition_string(starts_at_attr, ends_at_attr)
     primary_key_name = primary_key(record)
+    key = primary_key_value(primary_key_name, record)
     if record.new_record?
       self.sql_conditions = main_condition
     else
-      self.sql_conditions = "#{main_condition} AND #{record_table_name(record)}.#{primary_key_name} != #{record.send(primary_key_name)}"
+      if key.is_a?String
+        self.sql_conditions = "#{main_condition} AND #{record_table_name(record)}.#{primary_key(record)} != '#{key}'"
+      else
+        self.sql_conditions = "#{main_condition} AND #{record_table_name(record)}.#{primary_key(record)} != #{key}"
```

Comment:
Nope, the bug we found was for uuid #{key} needs to be wrapped in quotes otherwise it's treated like an integer and throws an error.